lib/NonlinearSolveBase/src/solve.jl

```diff
@@ -264,11 +314,20 @@ end
 
 function SciMLBase.__init(
         prob::AbstractNonlinearProblem, alg::AbstractNonlinearSolveAlgorithm, args...;
+        initializealg = NonlinearSolveDefaultInit(),
         kwargs...
 )
-    return NonlinearSolveNoInitCache(prob, alg, args, kwargs)
+    cache = NonlinearSolveNoInitCache(
+        prob, alg, args, kwargs, initializealg, ReturnCode.Success)
```

Comment:
```suggestion
        prob, alg, args, kwargs, initializealg, ReturnCode.Default)
```

Probably should be this